com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -385,6 +392,81 @@ public AsyncOperation UnloadSceneAsync(Scene scene, SceneEventProgress sceneEven
         /// instances with client unique scene instances
         /// </summary>
         internal Dictionary<int, int> ServerSceneHandleToClientSceneHandle = new Dictionary<int, int>();
+        internal Dictionary<int, int> ClientSceneHandleToServerSceneHandle = new Dictionary<int, int>();
+        internal Dictionary<int, Scene> HandleToScene = new Dictionary<int, Scene>();
+
+        /// <summary>
+        /// Add the client to server (and vice versa) scene handle lookup.
+        /// Add the client-side handle to scene entry in the HandleToScene table.
+        /// If it fails (i.e. already added) it returns false.
+        /// </summary>
+        internal bool UpdateServerClientSceneHandle(int serverHandle, int clientHandle, Scene localScene)
+        {
+            if (!ServerSceneHandleToClientSceneHandle.ContainsKey(serverHandle))
+            {
+                ServerSceneHandleToClientSceneHandle.Add(serverHandle, clientHandle);
+            }
+            else
+            {
+                return false;
+            }
+
+            if (!ClientSceneHandleToServerSceneHandle.ContainsKey(clientHandle))
+            {
+                ClientSceneHandleToServerSceneHandle.Add(clientHandle, serverHandle);
+            }
+            else
+            {
+                return false;
+            }
+
+            if (!HandleToScene.ContainsKey(clientHandle))
+            {
+                HandleToScene.Add(clientHandle, localScene);
+            }
+            else
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Removes the client to server (and vice versa) scene handles.
+        /// If it fails (i.e. already removed) it returns false.
+        /// </summary>
+        internal bool RemoveServerClientSceneHandle(int serverHandle, int clientHandle)
+        {
+            if (ServerSceneHandleToClientSceneHandle.ContainsKey(serverHandle))
+            {
+                ServerSceneHandleToClientSceneHandle.Remove(serverHandle);
+            }
+            else
```

Comment:
Not sure what the process on errors/asserts is but it might be worth making this fall through and then raise asserts or errors or warnings when ClientSceneHandleToServerSceneHandle and HandleToScene contain members not added via UpdateServerClientSceneHandle.  I'm assuming here that these Dictionaries can only be modified by these two functions.  Actually if thats the case then it might be better to group the containers into a class/struct, have the add/remove function and then just expose the accessors, that way you can be sure no one is going to directly modify the dictionaries in the future (makes the grouping of these containers explicit to anyone working on this code in the future, rather than implicit via these two functions)

Reply:
That is a really good point.  Let me look at this again, but I think your suggestion is a good idea.

Reply:
This area has several tendrils in various areas. I like the idea, but I am going to resist making the change in this PR as it should really be done as a separate PR.